hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/spark/sql/hudi/TestInsertTable.scala

```diff
@@ -2160,6 +2174,8 @@ class TestInsertTable extends HoodieSparkSqlTestBase {
            |union
            |select '1' as id, 'aa' as name, 123 as dt, '2023-10-12' as `day`, 12 as `hour`
            |""".stripMargin)
+      val stageClassName = classOf[HoodieSparkEngineContext].getSimpleName
+      spark.sparkContext.addSparkListener(new StageParallelismListener(stageName = stageClassName))
```

Comment:
@xuzifu666 Can you have StageParallelismListener update a shared counter (static) and assert here that the count increased by atleast one to ensure StageParallelismListener was indeed called as expected ?

Reply:
@bvaradar OK，had add a counter to assert called atleast once. PTAL

Reply:
@bvaradar Had add assert to make sure be called atleast once，PTAL